apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx

```diff
@@ -361,6 +373,30 @@ export class WorkshopForm extends React.Component {
     return this.state.course && Subjects[this.state.course];
   }
 
+  // For summer workshops for CSP and CSA, only workshop admins can change
+  // the “Is this a virtual workshop” field within one month of the workshop
+  // in order to prevent people from changing this last minute without
+  // talking to us or the Friday institute.
+  checkCannotChangeIfWorkshopVirtual(sessions, course) {
+    return (
+      (course === COURSE_CSP || course === COURSE_CSA) &&
+      !this.props.permission.has(WorkshopAdmin) &&
+      sessions.some(this.workshopStartsWithinMonth)
+    );
+  }
+
+  // Returns whether today is within a month before the workshop start.
+  workshopStartsWithinMonth(session) {
+    const today = new Date();
+    const workshopDate = new Date(session.date);
+    const monthBeforeWorkshopDate = new Date(
+      workshopDate.getFullYear(),
+      (workshopDate.getMonth() + 11) % 12,
```

Comment:
Instead of doing (month - 1) to go a month before I did '+ 11 % 12' to get the same result but avoid negative numbers when representing months.